prov/gni/configure.m4

```diff
@@ -9,6 +9,13 @@ AC_DEFUN([FI_GNI_CONFIGURE],[
         # have to pull in pkg.m4 manually
 	ugni_lib_happy=0
 	gni_header_happy=0
+	GNI_GCC_VERSION="4.9.0"
+	# Pull out gcc version number with regex backreference
+	gcc_version=`gcc --version | head -n1 | sed 's/.*\([[0-9]]\.[[0-9]]\.[[0-9]]\).*/\1/'`
+	if [[ "$gcc_version" \< "$GNI_GCC_VERSION" ]]; then
+	   AC_MSG_ERROR([gni provider requires gcc version "$GNI_GCC_VERSION" or higher but gcc version "$gcc_version" is being used.])
```

Comment:
To fix the problem with travis/os-x test, switch to using ```AC_MSG_WARN```,
then ugni_lib_happy=0.

But this whole block should e moved to below line 45 so that the compiler check is the last thing after all of the alps, ugni checks.